src/org/rust/lang/structure/RustFileTreeElement.kt

```diff
@@ -18,6 +16,14 @@ class RustStructureViewTreeElement(file: RustFileImpl) : PsiTreeElementBase<Rust
                 o.acceptChildren(this)
             }
 
+            override fun visitEnumItem(o: RustEnumItem) {
+                result += (RustStructureViewElement(o))
```

Comment:
Parenthesis are not neccesary, they were an artifact of refactorig :)

You could use https://github.com/alexeykudinkin/intellij-rust/blob/master/src/org/rust/lang/core/psi/util/RecursiveRustVisitor.kt here.